hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java

```diff
@@ -782,6 +795,89 @@ public ScmInfo getScmInfo() throws IOException {
     }
   }
 
+  @Override
+  public void transferLeadership(String newLeaderId)
+      throws IOException {
+    getScm().checkAdminAccess(getRemoteUser());
+    if (!SCMHAUtils.isSCMHAEnabled(getScm().getConfiguration())) {
+      throw new SCMException("SCM HA not enabled.", ResultCodes.INTERNAL_ERROR);
+    }
+    boolean auditSuccess = true;
+    final Map<String, String> auditMap = Maps.newHashMap();
+    auditMap.put("newLeaderId", newLeaderId);
+    try {
+      SCMRatisServer scmRatisServer = scm.getScmHAManager().getRatisServer();
+      RaftGroup group = scmRatisServer.getDivision().getGroup();
+      RaftPeerId targetPeerId;
+      if (newLeaderId.isEmpty()) {
+        RaftPeer curLeader = ((SCMRatisServerImpl) scm.getScmHAManager()
+            .getRatisServer()).getLeader();
+        targetPeerId = group.getPeers()
+            .stream()
+            .filter(a -> !a.equals(curLeader)).findFirst()
+            .map(RaftPeer::getId)
+            .orElseThrow(() -> new IOException("Cannot" +
+                " find a new leader to transfer leadership."));
+      } else {
+        String ratisAddr = convertToRatisAddr(newLeaderId);
```

Comment:
Use `RaftPeerId.getRaftPeerId(newLeaderId)` to get `RaftPeerId`; see https://github.com/apache/ozone/blob/master/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMRatisServerImpl.java#L172

Reply:
Hi @szetszwo , SCM is different from OM as it uses UUID as raftPeerId, See

```
  public String getScmId() {
    return getScmStorageConfig().getScmId();
  }
```

https://github.com/apache/ozone/blob/e7d57b9f911ca9878f2255cf79d8b23b9c2562f3/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMStorageConfig.java#L90-L98

Reply:
> ... as it uses UUID as raftPeerId, ...

What is the problem of using UUID?  Too long?

If we want to support passing an address, we should have a `--newLeaderAddress` option.

Reply:
> What is the problem of using UUID? Too long?


Hi @szetszwo.
OM uses the OmNodeId in the conf as the **RaftPeerId**, so we can directly find the new leader in the Raft group info by the passed nodeId.
SCM uses the UUID as the **RaftPeerId**, we have no way to know the new leader's UUID, so I added the conversion from ScmNodeId to RatisAddress and then got the **RaftPeerId**.

Reply:
@szetszwo , We have had scmId in the urrent Role info. The sideeffect is the different shell usage between OM and SCM, but seems not a big problem.  I have updated the code.